test/gtest/ucp/test_ucp_tag.cc

```diff
@@ -482,6 +592,59 @@ UCS_TEST_P(test_ucp_tag_limits, check_max_short_zcopy_thresh_zero, "ZCOPY_THRESH
               ucp_ep_config(sender().ep())->tag.eager.zcopy_thresh[0]);
 }
 
+UCS_TEST_P(test_ucp_tag_limits, check_rndv_thresh,
+           "RNDV_THRESH=0")
+{
+    check_rndv_startup_config(0, 0);
+    check_rndv_threshold(0);
+}
+
+UCS_TEST_P(test_ucp_tag_limits, check_rndv_intra_thresh,
+           "RNDV_THRESH=auto,intra:20")
+{
+    check_rndv_startup_config(20, UCS_MEMUNITS_AUTO);
+    check_rndv_threshold(20);
+}
+
+UCS_TEST_P(test_ucp_tag_limits, check_rndv_intra_thresh_large,
+           "RNDV_THRESH=auto,intra:2000")
+{
+    check_rndv_startup_config(2000, UCS_MEMUNITS_AUTO);
+    check_rndv_threshold(2000);
+}
+
+UCS_TEST_P(test_ucp_tag_limits, check_rndv_intra_thresh_inf,
+           "RNDV_THRESH=auto,intra:inf")
+{
+    check_rndv_startup_config(UCS_MEMUNITS_INF, UCS_MEMUNITS_AUTO);
+    /* check that rndv protocol is disabled */
+    check_ep_proto_rndv(UCS_MEMUNITS_INF, false);
+}
+
+UCS_TEST_P(test_ucp_tag_limits, check_rndv_intra_thresh_common,
+           "RNDV_THRESH=10,intra:20")
+{
+    check_rndv_startup_config(20, 10);
+    check_rndv_threshold(20);
+}
+
+UCS_TEST_P(test_ucp_tag_limits, check_rndv_inter_thresh,
+           "RNDV_THRESH=auto,inter:30")
+{
+    check_rndv_startup_config(UCS_MEMUNITS_AUTO, 30);
+    /* TODO: configure/mock inter-node in test */
+
+    /* check that inter-node config is ignored for intra-node */
+    check_ep_proto_rndv(30, false);
+}
+
+UCS_TEST_P(test_ucp_tag_limits, check_rndv_inter_thresh_common,
+           "RNDV_THRESH=1000,inter:30")
+{
+    check_rndv_startup_config(1000, 30);
+    check_rndv_threshold(1000);
+}
```

Comment:
As I can see almost all the tests constains the same scenario:
1. Set RNDV thresholds environement
2. check_rndv_startup_config
3. check_rndv_threshold

WDYT, would it has sense to unify all this checks under one test and do something like:
```c++
struct rndv_thresh {
    size_t inter;
    size_t intra;
    const char *string;
}

UCS_TEST_P(test_ucp_tag_limits, check_rndv_thresh) {
    std::vector<rndv_thresh> env_variants = {
        {0, 0, 0, "0"},
        {UCS_MEMUNITS_AUTO, 20, UCS_MEMUNITS_AUTO, "intra:20"},
        ....
    }

    for (const auto& env: env_variants) {
         UCS_TEST_MESSAGE << " Check RNDV thresholds for UCX_RNDV_THRESH=" << env.string;
         ucp_config_modify(config, "RNDV_THRESH", env.string);
         check_rndv_startup_config(env.intra, env.inter);
         check_rndv_threshold(env.intra);
    }
}
```

Reply:
Mmm, honestly I wouldn't go this way for 2 reasons:

1. Apart from the function calls that you mentioned, each test performs initialization part which is not trivial: look at `test_ucp_tag_limits::init`. It creates sender and receiver, init eps, connect them...
With your approach we do this init only once, and then just change the config.
But I doubt that changing config affects already created eps. ?
If so, then we cannot test runtime config

2. Even if p1 worked out, personally I would still prefer to have these tests independent of each other, for debugability reason.

Reply:
I think the first point can be potentially solved by explicit `test::init()` and `test::cleanup()` calls, but that can be complicated....

Ok, up to you :)